### PR TITLE
add buy-plank support for menu entry swapper

### LINKED ISSUE #5526: Add the Buy Plank option to the menu entry swapper
Basically, It would help me a lot (and others doing the make plank money maker) if someone could swap the menu entry boxes for "Buy-plank" with the "Trade" on the sawmill operator.

### PR DESCRIPTION
**Pull request by astaninger**
Fix #5526

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java [hunk 1]
```diff
@@ -485,6 +490,7 @@ else if (config.swapBones() && option.equals("bury"))
 		{
 			swap("use", option, target, true);
 		}
+
```
Comment (Gamer1120) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java]: Remove newline

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java [hunk 2]
```diff
@@ -239,4 +239,15 @@ default boolean swapDecant()
...
 		return false;
 	}
+
+	@ConfigItem(
+			position = 19,
+			keyName = "swapBuy-plank",
+			name = "Buy-plank",
+			description = "Swap Talk-to with Buy-Plank for Sawmill Operator at the LumberYard."
+	)
+	default boolean swapPlank()
```
Comment (Gamer1120) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java]: Change this to swapBuyPlank

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java [hunk 3]
```diff
@@ -362,6 +362,11 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 				swap("assignment", option, target, true);
 			}
 
+			if (config.swapPlank())
```
Comment (Gamer1120) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java]: Change this to swapBuyPlank

**Comment by Gamer1120** [comment]
Commits should also be squashed.

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java [hunk 4]
```diff
@@ -239,4 +239,15 @@ default boolean swapDecant()
 	{
 		return false;
 	}
+
+	@ConfigItem(
+			position = 19,
+			keyName = "swapBuy-plank",
+			name = "Buy-plank",
+			description = "Swap Talk-to with Buy-Plank for Sawmill Operator at the LumberYard."
```
[the remark this answers is not in the document]
Comment (astaninger) [reply runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java]: It's not capitalized

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java [hunk 5]
```diff
@@ -239,4 +239,15 @@ default boolean swapDecant()
 	{
 		return false;
 	}
+
+	@ConfigItem(
+			position = 19,
+			keyName = "swapBuy-plank",
```
[the remark this answers is not in the document]
Comment (raiyni) [reply runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java]: If Buy-plank is hyphenated, I think the pedantic name as a key is fine.

**Comment by Gamer1120** [comment]
Oh, you're right. It is called Lumber Yard. That's interesting. Especially since the scroll that teleports there is called "Lumberyard teleport". Jagex being consistent :P

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java [hunk 6]
```diff
@@ -239,4 +239,15 @@ default boolean swapDecant()
 	{
 		return false;
 	}
+
+	@ConfigItem(
+			position = 19,
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java]: Fix the indentation

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java [hunk 7]
```diff
@@ -239,4 +239,15 @@ default boolean swapDecant()
...
+
+	@ConfigItem(
+			position = 19,
+			keyName = "swapBuyPlank",
+			name = "Buy-plank",
+			description = "Swap Talk-to with Buy-Plank for Sawmill Operator at the Lumber Yard."
+	)
+	default boolean swapBuyPlank() 
+	{ 
+		return false;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java]: This seems like good default to be true? Not sure, never made planks myself

Comment (Gamer1120) [reply runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java]: Trade is the default when you have the trade menu entry swapper setting on.

**Comment by astaninger** [comment]
wait how do i squash

**Comment by Alexsuperfly** [comment]
GH has a squash button for merging the PR, so its not necessary if theres only going to be one commit

**Comment by raiyni** [comment]
Squashing is mainly important if you are going to have merge conflicts

**Comment by ollieatkinson** [comment]
> wait how do i squash

You can use an interactive rebase then squash all the commits into the root

```
$ git rebase --interactive --root
```

https://dev.to/blakedeboer/beginners-guide-to-interactive-rebasing-1ob

**Comment by melkypie** [comment]
This was added in menu swapper extended in the plugin hub

**Pull request closed without merging**